### ISSUE TITLE
consider sourcing releases from modrinth (in the future)

### ISSUE THREAD

**Issue by plexiondev** [plexiondev/plexiondev.github.io#182]
- depends on #128 
- related to #168 
- related to #157 

with modrinth's upcoming datapack support, it offers some exclusive features and metadata not available currently on github, including:

- **downloading as a fabric/forge/quilt packaged mod**
- minecraft version
- designated primary file download
- designated 'featured' flag

only thing holding this back (for older projects) is lack of support for dating back older versions/re-ordering - but this is apparently planned.

**Comment by plexiondev** [comment]
sample response for reference:
```json
{
    "id": "JSWCXPl0",
    "project_id": "GRBFvrqx",
    "author_id": "VG0sMqj1",
    "featured": true,
    "name": "Fluorite",
    "version_number": "2022.0927",
    "changelog": "A complete redesign of every aspect in vanilla Minecraft. This has been under construction for almost a year now and it's finally here.\n\nCompared to the original Simple Dark, elements are a lot more contrasted, in combination with shadows making elements easier to see at a glance (something the original could struggle at). Due to this, the Contrasted release will most likely be deprecated in the near future, it has not been updated in this redesign and is only attached here now for compatiability reasons.\n\n![2022-09-27_18 26 53](https://user-images.githubusercontent.com/46572320/192604316-39314785-5e90-4249-baf7-952c76f947c8.png)\n![2022-09-27_18 27 03](https://user-images.githubusercontent.com/46572320/192604226-3154b8d2-f5ab-4dee-b51c-7c5e7bc0ff35.png)\n![2022-09-27_18 26 25](https://user-images.githubusercontent.com/46572320/192604240-949d8602-db34-44e3-acfa-f54acfb21595.png)\n![2022-09-27_18 30 30](https://user-images.githubusercontent.com/46572320/192604397-a9bb8ddd-d14e-4fdf-81ed-14103fb41eba.png)\n\n#### Interfaces\n- Completely redesigned all interfaces\n- Fixed inconsistencies and overall bad design choices in various places<p class=\"text-14 addition\">This pack has existed since January 2021, so it makes sense for there to be inconsistencies.</p>\n\n#### Season-variation\n- Season-themed releases (eg. Halloween and Christmas) will arrive at a later date<p class=\"text-14 addition\">Probably closer to Halloween.</p>\n\n#### Mod support\n- Mod support will come at a later date, for now it uses the original Simple Dark look<p class=\"text-14 addition\">It is going to take **a lot** of work to redesign all mod interfaces, but i'll get to it eventually.</p>\n- Along with this, Realms will also remain with the original Simple Dark look for now",
    "changelog_url": null,
    "date_published": "2022-09-27T18:44:50.117345Z",
    "downloads": 52,
    "version_type": "release",
    "files": [
      {
        "hashes": {
          "sha512": "38165845f6cc20b7d3efd19cfbfe34838c1458539845a75c50dcdd9c64c33a9cea4d2f3d5cfcd2ab93bdb12ad12ae3c65e8638c5e5751c9f7d3f2914e412e8a9",
          "sha1": "8fffab1701ea8883289c78e7a1da1dbc5c0c9c32"
        },
        "url": "https://cdn.modrinth.com/data/GRBFvrqx/versions/JSWCXPl0/Simple%20Dark.zip",
        "filename": "Simple Dark.zip",
        "primary": true,
        "size": 1247331
      }
    ],
    "dependencies": [],
    "game_versions": [
      "1.16",
      "1.16.1",
      "1.16.2",
      "1.16.3",
      "1.16.4",
      "1.16.5",
      "1.17",
      "1.17.1",
      "1.18",
      "1.18.1",
      "1.18.2",
      "1.19",
      "1.19.1",
      "1.19.2"
    ],
    "loaders": [
      "minecraft"
    ]
  }
```

**Comment by plexiondev** [comment]
the new plan is to retrieve github api releases as usual but also modrinth api releases. from there, if version numbers match between both::

- use github changelog data
- use modrinth metadata (big)
- attempt to combine download count if possible?
